### ISSUE TITLE
FutureWarnings when using `apply_value_format` in `read_sav` 

### ISSUE THREAD

**Issue by ME-researchgroup** [Roche/pyreadstat#252]
Hi!

Just want to give a heads up that the `apply_value_formats` argument to `read_sav` will break in a future version of pandas, if you hadn't noticed already.

```  df, meta = pyreadstat.read_sav("test_dataset.sav", apply_value_formats=True)
[...]\test.py:4: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
A typical example is when you are setting values in a column of a DataFrame, like:

df["col"][row_indexer] = value

Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.    

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy```

**Comment by stefanverbeeck** [comment]
I want to add that this futurewarning also shows up when reading sas7bdat or xpt, which according to the docs needs to be done with a chained assignment like this:
df, meta = ps.read_sas7bdat(...)
Still works fine anyway.

**Comment by ofajardo** [comment]
Thanks for the report. I ackowledge the problem and will work on it soon. I see plenty of warnings when I run the tests, so a lot to clean.

**Comment by ofajardo** [comment]
yes, pandas 3.0 will for sure break pyreadstat. Let's see how painful it is to fix it, these warnings already have been painful. But on the long run it is going to be better. 

The warnings are now fixed in version 1.2.7, so closing this. Let me know if you find more in the future!

btw, cpython is what you are using when you use pyreadstat (in contrast to Jython, iron-python, pypy ...). Maybe you meant cython =)
